garden_ai/pipelines.py

```diff
@@ -77,6 +84,70 @@ def check_steps_composable(cls, steps):
             raise
         return steps
 
+    @validator("requirements_file")
+    def check_exists(cls, req_file):
+        req_file = pathlib.Path(req_file)
+        if not req_file.exists():
+            raise ValueError(f"Could not find {req_file}. Please make sure it exists.")
+        return str(req_file)
+
+    @validator("requirements_file")
+    def check_extension(cls, req_file):
+        if not req_file.endswith((".txt", ".yml", ".yaml")):
+            raise ValueError(
+                "Did not recognize requirements file extension. Try"
+                "again with a standard requirements.txt or conda"
+                "environment.yml/yaml."
+            )
+        return req_file
+
+    def _collect_requirements(self):
+        """collect requirements to pass to funcx container service.
+
+        Populates attributes: ``self.python_version, self.pip_dependencies, self.conda_dependencies``
+        """
+
+        # mapping of python-version-witness: python-version (for warning msg)
+        py_versions = {
+            "system": ".".join(map(str, sys.version_info[:3])),
+            "pipeline": self.python_version,
+        }
+        if self.requirements_file.endswith((".yml", ".yaml")):
+            py_version, conda_deps, pip_deps = read_conda_deps(self.requirements_file)
+            if py_version:
+                py_versions["pipeline"] = py_version
+            self.conda_dependencies += conda_deps
+            self.pip_dependencies += pip_deps
+
+        elif self.requirements_file.endswith(".txt"):
+            with open(self.requirements_file, "r") as f:
+                contents = f.read()
+                parsed = dparse.parse(
+                    contents, path=self.requirements_file, resolve=True
+                ).serialize()
+                deps = [d["line"] for d in parsed["dependencies"]]
+                deps.extend(d["line"] for d in parsed["resolved_dependencies"])
+                self.pip_dependencies += deps
+
+        for step in self.steps:
+            self.conda_dependencies += step.conda_dependencies
+            self.pip_dependencies += step.pip_dependencies
+            py_versions[step.__name__] = step.python_version
+
+        self.python_version = py_versions["pipeline"] or py_versions["system"]
+        self.conda_dependencies = list(set(self.conda_dependencies))
+        self.pip_dependencies = list(set(self.pip_dependencies))
+
+        if len(set(py_versions[k] for k in py_versions if py_versions[k])) > 1:
+            logger.warning(
+                "Found multiple python versions specified across this"
+                f"pipeline's dependencies: {py_versions}. {self.python_version} "
```

Comment:
It seems like this warning would fire if the user's system python was different from the version they explicitly specified for their pipeline, even if the pipeline and its steps are all on the same page. Do we want that behavior?

Reply:
I have a feeling that more info is better here, especially since a 'system' version that isn't as expected could be causing problems with un-/pickling, or could be the case that it's been used to infer something by mlflow incorrectly. 

This is also why I opted to not raise any exceptions, just logging a warning instead